### ISSUE TITLE
creating untitled exchange accounts after setup is done

### ISSUE THREAD

**Issue by MacBucky** [talkingmoose/Outlook-Exchange-Setup-5#10]
I have found the scripts to be working as designed and everything looks good initially. However, I had a couple of users now report when opening outlook the next day, there is now an untitled account listed in the accounts window.  These untitled accounts are actually the user's account, just without a name and titled, untitled.

This is having a negative affect in that, calendars are showing multiple occurrences of the same item.  In one case, she had four pop ups for the same meeting at 9 AM.  Once I cleared out the untitled accounts, everything returned to normal.  However, I cannot guarantee that this won't begin occurring again on Monday.

I saw on the JAMFNation forums, if the kerberos section of the AppleScript was set to false, this might happen.  So, on one Mac yesterday, I changed it to true.  This morning Susan had an error saying kerberos wasn't enabled in our environment, click okay to disable kerberos.

It is appearing or acting, as if the script is running over and over on each subsequent restart.  If this is the case, which script should I remove in order to remedy this?

Thank you,

Mick

**Comment by ygini** [comment]
Same behavior here.

**Comment by talkingmoose** [comment]
I've just updated the shell script that creates the launch agent in the user's folder to include more logging. If the script is creating a new account with each launch of Outlook then:
1. Either the AppleScript is not finishing correctly and not deleting the launch agent in ~/Library/LaunchAgents, or...
2. The shell script in /Library/Talking Moose Industries/Scripts is not correctly identifying the correct files and folders exists or don't exist.

Once an Exchange account has been created, the log file should report at each login:

"/Users/mmoose/Library/Group Containers/UBF8T346G9.Office folder already exists. Doing nothing."
"/Users/mmoose/Group Containers/UBF8T346G9.Office/OutlookProfile.plist already exists. Doing nothing."
"/Users/mmoose/Library/LaunchAgents already exists. Doing nothing."
"/Users/mmoose/Library/LaunchAgents/net.talkingmoose.OutlookExchangeSetup5.plist does not exist. Doing nothing."

If any of the above lines returns "Something may be wrong" then the specific folder or file either exists when it shouldn't or doesn't exist when it should.

**Comment by csteelatgburg** [comment]
Posting here for completeness:
I had two versions of the LaunchAgent in /Library/LaunchAgents and they were getting in each others way. I removed the older version and I no longer get multiple Exchange accounts being configured.

**Comment by talkingmoose** [comment]
For anyone who's seeing either of these issues...
1. Outlook automatically launching when a user logs in
2. Multiple accounts getting configured

Verify that only this file exists in the top-level /Library/LaunchAgents folder:
net.talkingmoose.OutlookExchangeSetupLaunchAgent.plist

Other net.talkingmoose files would be cruft from earlier versions. I will update the installer script to check for and remove those.

**Comment by MacBucky** [comment]
Thanks Bill,

I am seeing two versions in the /Library/LaunchAgents folder.  So, I am going to repackage this again and see if this is resolved.  I am little perplexed how the older version 5 was there to begin with, but knowing this, I am going to manually do this on another Mac that hasn’t been touched yet.

Mick

**Comment by MacBucky** [comment]
I just followed the steps to package again using the Package for Deployment AppleScript.  It did in fact create two plist files in the /Library/LaunchAgents folder.  One is the one indicated below and there is another one called net.talkingmoose.OutlookExchangeSetup5.plist

So I think that is the problem.

I am going to move the files into the appropriate locations and package using Composer to grab them.

Mick

**Comment by talkingmoose** [comment]
@MacBucky, that's very strange. This file:

`net.talkingmoose.OutlookExchangeSetup5.plist`

is not part of the folder structure to make the package. It's created on the fly by the shell script when the user logs in.

I just downloaded the 5.3.0 set of file and ran the Package for Deployment AppleScript without making any modifications and it packaged correctly for me. I'm using Suspicious Package to view the contents of the package it created.

![package](https://cloud.githubusercontent.com/assets/877471/14646835/bdc78b5e-0620-11e6-9e75-c573b6cfff1b.png)

**Comment by MacBucky** [comment]
I have no idea why this would be happening either.  This is on a base iMac with the bare minimum software installed and without any office products.  I will verify that this is ALL that is being installed, but like I said, very strange.

**Comment by MacBucky** [comment]
Here is the screen shot of the main folders prior to packaging up using the applescript.

Also, I looked PRIOR to making this package and there isn’t a thing in the /Library/LaunchAgents folder for purposes of the talking moose stuff.  Not sure what is happening here.

Mick

**Comment by talkingmoose** [comment]
@MacBucky, I think your attachment got stripped from the reply again. You probably need to post via the GitHub page.

Package your files and then use something like Suspicious Package after the fact to view the contents. That'll be very telling.

**Comment by MacBucky** [comment]
Thanks Bill.

Put it this way, using suspicious package, mine shows exactly the same as yours.  However, when I run this package on a fresh Mac, it actually deploys the additional file that is causing the problem.

**Comment by talkingmoose** [comment]
@MacBucky, a package can't deploy a file that's not in it.

I would **_fully expect_** the second launch agent to appear, though, on a fresh Mac the first time you log in. Do you think that's what you're seeing?

After you launch and configure Outlook on the fresh Mac, log out and log in again. On any subsequent login, does Outlook launch automatically (it shouldn't) or do you have any problems with additional accounts getting created?

**Comment by MacBucky** [comment]
Thanks Bill, that certainly could be the visual I am seeing.  Testing again on a fresh image and will let you know the results shortly.

Mick

**Comment by MacBucky** [comment]
Well initially, the scripts have auto launched Outlook BEFORE I opened it.  I am now restarting to see if additional accounts are being created.  As suspected Bill, the additional untitled accounts are still being created on each log into Outlook.

Now, with this being said, the two Launch agent files are still present and they are not being deleted.  This is on a base El Capitan Mac with Office 2016 being installed via self service.

Mick

**Comment by MacBucky** [comment]
Well here is what I have found.

If I use the JSS to create a quick script to remove that additional launch agent item, everything works fine.  So with my installation of Office 2016, I use everything as is, even as seen below for the setup package.  Before I run anything though, I remove that one additional item via a script that runs AFTER everything else.  I restart and everything works great.

So somehow Bill, that additional item is being placed in the launch agents folder.  Not sure why but if I remove it before restarting the Mac, everything is fine.

Mick

**Comment by grobin89** [comment]
Hey, I've been troubleshooting this issue for a while. In my scenario, I am deploying the package via JSS.

So I've found that on initial deployment, the Launchd `net.talkingmoose.OutlookExchangeSetupLaunchAgent.plist` runs. The launch agent runs `OutlookExchangeSetupLaunchAgent.sh` as root instead of the current user. Since the shell script runs as root, all directories and files are created under roots home directory, i.e `/`, instead of the current users home directory. It becomes more evident when the log is created at `/Library/Logs/OutlookExchangeSetup5.log` and the paths specified are listed under root. So, `$HOME = /`.

When launching Outlook for the first time, `Outlook Exchange Setup 5.2.1.scpt` launches as expected and creates the users account. It appears that the apple script runs as the current user and the log is created as expected at `~/Library/Logs/OutlookExchangeSetup5.log`.

To compare here are the two logs:
**~/Library/Logs/OutlookExchangeSetup5.log**

```
22/04/2016 11:18:44 am  Starting Exchange account setup...
22/04/2016 11:18:44 am  Script: Outlook Exchange Setup 5.2.1

22/04/2016 11:18:44 am  Setup properties...
22/04/2016 11:18:44 am  Use Kerberos: false
22/04/2016 11:18:44 am  Exchange Server: xxxxx
22/04/2016 11:18:44 am  Exchange Server Requires SSL: true
22/04/2016 11:18:44 am  Exchange Server Port: 443
22/04/2016 11:18:44 am  Directory Server: xxxxx
22/04/2016 11:18:44 am  Directory Server Requires Authentication: true
22/04/2016 11:18:44 am  Directory Server Requires SSL: false
22/04/2016 11:18:44 am  Directory Server SSL Port: 3268
22/04/2016 11:18:44 am  Directory Server Maximum Results: 1000
22/04/2016 11:18:44 am  Directory Server Search Base:
22/04/2016 11:18:44 am  Get User Information from Active Directory: true

22/04/2016 11:18:44 am  Verify Email Address: false
22/04/2016 11:18:44 am  Verify Server Address: false
22/04/2016 11:18:44 am  Display Domain Prefix: false
22/04/2016 11:18:44 am  Download Headers Only: false
22/04/2016 11:18:44 am  Hide On My Computer Folders: false
22/04/2016 11:18:44 am  Unified Inbox: false
22/04/2016 11:18:44 am  Disable Autodiscover: false
22/04/2016 11:18:44 am  Error Message text: Outlook's setup for your Exchange account failed. Please contact the Help Desk for assistance.

22/04/2016 11:18:45 am  User information...
.... I took this part out to make it shorter

22/04/2016 11:18:46 am  Set Group Similar Folders to false: Successful.
22/04/2016 11:18:46 am  Set Hide On My Computer Folders to false: Successful.
22/04/2016 11:18:47 am  Create Exchange account: Successful.
22/04/2016 11:18:47 am  Populate Me Contact information: Successful.
22/04/2016 11:18:47 am  Set Outlook as default mail, calendar and contacts application: Successful.
22/04/2016 11:18:47 am  Delete OutlookExchangeSetup5.plist file from user LaunchAgents folder: Failed.
```

**/Library/Logs/OutlookExchangeSetup5.log**

```
4/26/16 9:38:11 AM  Folder "/Library/Group Containers/UBF8T346G9.Office" does not exist.
4/26/16 9:38:11 AM  Create folder "/Library/Group Containters/UBF8T346G9.Office": Successful.
4/26/16 9:38:11 AM  Create empty file "/Library/Group Containers/UBF8T346G9.Office/OutlookProfile.plist": Successful.
4/26/16 9:38:11 AM  Create folder "/Library/LaunchAgents": Successful.
4/26/16 9:38:11 AM  Create launch agent "/Library/LaunchAgents/net.talkingmoose.OutlookExchangeSetup5.plist": Successful.
4/26/16 9:38:11 AM  Set launch agent permissions to 644 (-rw-r--r--): Successful.
4/26/16 9:38:11 AM  Load launch agent: Successful.
```

Hopefully this explains why the mysterious `OutlookExchangeSetup5.plist` is showing up in `/Library/LaunchAgents`. I'm not sure how to fix it. I'll give @MacBucky work around a go.

**Comment by talkingmoose** [comment]
@grobin89, thank you for the very detailed report!

None of the processes should be running as root. Each should be running under the current user's login, which will create the files/folders within the user's space.

I'll review the scripts and see if I can determine the circumstances where they may run as root.

**Comment by talkingmoose** [comment]
After some testing, I've found a couple of unusual scenarios where two launch agents appear in the main /Library/LaunchAgents folder instead of one in the /Library/LaunchAgents folder and the other in the user's ~/Library/LaunchAgents folder. (Enable and log in to OS X's root account and install the package or install the package via command line after using sudo -s.)

Therefore, I modified the postinstall script to not run the user's launch agent as root if installing by either of those methods or something similar.

This should take care of some issues that may be causing multiple untitled accounts and/or launching Outlook automatically.

You'll find v5.4.0 here:
https://github.com/talkingmoose/Outlook-Exchange-Setup-5

And the change log here:
https://github.com/talkingmoose/Outlook-Exchange-Setup-5/wiki/Change-log
